vignettes/articles/oauth.Rmd

```diff
@@ -13,25 +13,62 @@ knitr::opts_chunk$set(
 library(httr2)
 ```
 
-OAuth[^1] is a authorization framework design for performing work on the behalf of a user.
-It allows you to wrap an API that requires a user login without your app ever directly seeing the user's user name and password.
-You've almost certainly used this before on the web or with your TV; it's used in most cases where one website (or your TV) needs to login to an account on your behalf.
+OAuth[^1] is an authorization framework design for performing work on the behalf of a user.
+You've probably used it a bunch without knowing what it's called: it's used when you login to a non-Google website using your Google account, when you give an phone access to your twitter account, or when you login to a streaming app on your smart TV.
 
 [^1]: Here I'll only talk about OAuth 2.0 which is the only version in common use today.
     OAuth 1.0 is largely only of historical interest.
 
-OAuth is a broad framework that has many different variants, called **flows**, which makes it hard to provide generalisable advice.
-The one commonality is that all flows eventually generate an **access token** (essentially a random string of numbers and letters) which is provided to the API in the form of a **bearer token** in the `Authorization` header.
-If you have this access token, you can use it directly with `req_auth_bearer_token()`, but in most cases you will want to give httr2 the information to generate and use the access token for you by calling a `req_oauth_` function.
-The rest of the vignette describes just how to do that.
+You'll notice none of these common scenarios quite match to using an R package to scrape data from a web API, and that's the pleasure and pain of OAuth.
+OAuth gives you the incredible power to extract data from popular web services, but it's fundamentally designed for a different use case.
+This means that while httr2 can make the interactive experience pretty seamless, unattended usage (i.e. on CI or in an automated script) is usually going to include some pain.
+
+This vignette builds on the techniques in discussed in `vignette("wrapping-apis")`, so I recommend you read that vignette first.
+I'll also assume that you're working in package, since most people want to wrap up an API into a bunch of R functions, but most of the ideas will also apply if you're creating a one off script.
+
+## OAuth basics
+
+OAuth is a broad framework that has many different variants, called **flows**, which makes it hard to provide sweeping generalisations, but the basic idea of OAuth is to create a hierarchy of increasing more specific and shorter lived credentials, so that the impact of a credential being lost is as small as possible.
```

Comment:
```suggestion
OAuth is a broad framework that has many different variants, called **flows**, which makes it hard to provide sweeping generalisations, but the basic idea of OAuth is to create a hierarchy of increasingly more specific and shorter-lived credentials, so that the impact of a credential being lost is as small as possible.
```